From 01479d79f1ea15642d87ee8d029f3c03bb642159 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Mon, 16 Sep 2019 19:24:15 +0200
Subject: [PATCH] Don't use thread safety annotations with gcc

These annotations are primarily useful for detecting race conditions in
code. They don't seem to have any runtime effect and are not supported
by gcc.

See: https://clang.llvm.org/docs/ThreadSafetyAnalysis.html
---
 base/include/android-base/thread_annotations.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/base/include/android-base/thread_annotations.h b/base/include/android-base/thread_annotations.h
index 53fe6dae9..bc62f80bd 100644
--- a/base/include/android-base/thread_annotations.h
+++ b/base/include/android-base/thread_annotations.h
@@ -18,7 +18,11 @@
 
 #include <mutex>
 
+#ifdef __clang__
 #define THREAD_ANNOTATION_ATTRIBUTE__(x) __attribute__((x))
+#else
+#define THREAD_ANNOTATION_ATTRIBUTE__(x)
+#endif
 
 #define CAPABILITY(x) \
       THREAD_ANNOTATION_ATTRIBUTE__(capability(x))
